testimony/__init__.py

```diff
@@ -92,190 +77,66 @@ def __init__(self, function_def, parent_class=None, testmodule=None):
-            module_docstring = module_docstring.decode('utf-8')
-        class_docstring = ast.get_docstring(self.parent_class_def)
-        if class_docstring and not isinstance(class_docstring, type(u'')):
-            class_docstring = class_docstring.decode('utf-8')
-        function_docstring = self.docstring
-        if function_docstring and not isinstance(
-                function_docstring, type(u'')):
-            function_docstring = function_docstring.decode('utf-8')
-        tags, unexpected_tags, _ = self._parse_tags(module_docstring)
-        class_tags, class_unexpected_tags, _ = self._parse_tags(
-            class_docstring)
-        function_tags, function_unexpected_tags, self.skipped_lines = (
-            self._parse_tags(function_docstring))
-
-        # Update context dictionaries
-        tags.update(class_tags)
-        tags.update(function_tags)
-        unexpected_tags.update(class_unexpected_tags)
-        unexpected_tags.update(function_unexpected_tags)
-
-        for tag, value in tags.items():
-            if tag == 'bz':
-                tag = 'bugs'
-            if tag == 'assert':
-                tag = 'assertion'
-            if tag == 'type':
-                tag = 'test_type'
-            setattr(self, tag, value)
-        self.unexpected_tags = unexpected_tags
+        # Parse module, class and function docstrings. Every loop update the
+        # already defined tokens and invalid_tokens. Doing in the mentioned
+        # order will ensure that function docstrings have more priority over
+        # class and module docstring respectively.
+        docstrings = [
+            ast.get_docstring(self.module_def),
+            ast.get_docstring(self.parent_class_def),
+            self.docstring
+        ]
+        for docstring in docstrings:
+            if docstring and not isinstance(docstring, type(u'')):
+                docstring = docstring.decode('utf-8')
+            tokens, invalid_tokens = self.parser.parse(docstring)
+            self.tokens.update(tokens)
+            self.invalid_tokens.update(invalid_tokens)
 
         # Always use the first line of docstring as test case name
-        if self.test is None:
-            self.test = self.docstring.strip().split('\n')[0]
-
-    @property
-    def automated(self):
-        """Indicate if the test case is automated or not."""
-        if self.status and self.status.lower() == 'manual':
-            return False
-        return True
+        if self.tokens.get('test') is None:
+            self.tokens['test'] = self.docstring.strip().split('\n')[0]
 
     @property
     def has_valid_docstring(self):
-        """Returns ``True`` if at least feature and assert tags are defined.
-        """
-        return (
-            self.assertion is not None and
-            self.feature is not None and
-            self.test is not None
-        )
+        """Boolean indicating if docstring have the minimum tokens."""
```

Comment:
To be consistent with other docstring:
Indicates if the docstring has the minimum tokens.